### PR TITLE
GH-#118 Server timeout when checking peers_nonce

### LINKED ISSUE #118: Server timeout when checking peers_nonoce
We recently introduced peers `nonce` check in `Aecore.Peers.Worker.add_peer`. The check was introduced to prevent node from adding itself to list of known peers by making `\info` request to peer that's about to be added and comparing it's nonce to own nonce. This doesn't work as expected, because `\info` requests uses `Aecore.Peers.Worker` so it can't be processed while `:add_peer` call to `Aecore.Peers.Worker` genserver is handled.

Error introduced by 2055e9fb (GH-79)

To reproduce:
```
iex(9)> Aecore.Peers.Worker.add_peer("http://localhost:4000") 
[info] GET /info
[debug] Processing with Aehttpserver.InfoController.info/2
  Parameters: %{}
  Pipelines: [:browser]
:error
iex(10)> [error] GET /info request error
[error] #PID<0.415.0> running Aehttpserver.Endpoint terminated
Server: localhost:4000 (http)
Request: GET /info
** (exit) exited in: GenServer.call(Aecore.Peers.Worker, :get_peers_nonce, 5000)
    ** (EXIT) time out

```

### PR DESCRIPTION
**Pull request by HristianHristov**
Resolves #118

### CODE REVIEW

File: apps/aecore/lib/aecore/peers/worker.ex [hunk 1]
```diff
@@ -69,23 +62,42 @@ defmodule Aecore.Peers.Worker do
...
+  def get_peer_nonce() do
+    case :ets.info(:nonce_table) do
+      :undefined -> create_nonce_table()
+      _ -> :table_created
+    end
+    case :ets.lookup(:nonce_table, :nonce) do
+      [] ->
+        rand = :rand.uniform(@mersenne_prime)
+        :ets.insert(:nonce_table, {:nonce, rand})
+        rand
```
Comment (cytadela8) [inline apps/aecore/lib/aecore/peers/worker.ex]: I would rename `rand` to `nonce` as it's a little bit confusing.

File: apps/aecore/lib/aecore/peers/worker.ex [hunk 2]
```diff
@@ -69,23 +62,42 @@ defmodule Aecore.Peers.Worker do
...
+    end
+  end
+
   ## Server side
 
   def init(initial_peers) do
     {:ok, initial_peers}
   end
 
-  def handle_call({:add_peer,uri}, _from, %{peers: peers, nonce: own_nonce} = state) do
```
Comment (cytadela8) [inline apps/aecore/lib/aecore/peers/worker.ex]: I would leave the the peers map wrapped in a state map. In future we might want to store more data in state.

**Comment by cytadela8** [comment]
I made the test that should have caught the problem earlier more precise.

File: apps/aecore/lib/aecore/peers/worker.ex (hunk 2 above)
Comment (thepiwo) [reply apps/aecore/lib/aecore/peers/worker.ex]: @HristianHristov will you adjust this as well?

**Pull request merged**